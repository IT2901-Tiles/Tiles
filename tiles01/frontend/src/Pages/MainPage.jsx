import React from 'react';
//Component imports
import Logo from "../Components/Logo";
import LinkButton from '../Components/LinkButton';
import CharNormalMainPage from "../Components/CharNormalMainPage";
//img imports
import img1 from "../images/img1.png"
import img2 from "../images/img2.png"
import {resetCards} from "../Components/AnimationsPreview"

function MainPage() {

    return (
        <div className="Page mainPage">
            <Logo/>
            <img className="img2" src={img2} alt="stack of cards2"/>
            <div className="infoBox">
                <p>Welcome to the Tiles Internet of Things Invention! <br/>
                Letâ€™s create some smart creations! </p>
            </div>
            <LinkButton title="Let's begin!" target="/cards" onClick={() => resetCards()} size="Large"></LinkButton>
            <img class="mainPage img1" src={img1} alt="stack of cards"/>
            <div id="charNormalMainWrapper">
                <CharNormalMainPage />
            </div>
        </div>
    );
}

export default MainPage;