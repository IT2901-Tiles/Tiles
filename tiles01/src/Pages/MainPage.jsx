import React from 'react';
//Component imports
import Logo from "../Components/Logo";
import LinkButton from '../Components/LinkButton';
//img imports
import motionCard from "../img/triggers/motion.png"
import colorChangeCard from "../img/feedback/colorChange.png"
import bikeCard from "../img/things/bike.png"
import dropCard from "../img/triggers/drop.png"

function MainPage() {

    return (
            <div className="Page mainPage">
                <div>
                    <img className="motionCardTop"  alt="The motion card" src={motionCard} />
                    <img className="colorChangeCardTop" alt="ColorChangeCard" src={colorChangeCard}/>
                    <img className="dropCardTop" alt="Drop card" src={dropCard} />
                </div>
                <Logo/>
                <div className="infoBox">
                    <p>Welcome to the Tiles Internet of Things Invention! <br/>
                    Letâ€™s create some smart creations! </p>
                </div>
                <LinkButton title="Let's begin!" target="/cards" size="Large"></LinkButton>
                <div>
                    <img className="motionCard"  alt="The motion card" src={motionCard} />
                    <img className="colorChangeCard" alt="ColorChangeCard" src={colorChangeCard}/>
                    <img className="bikeCard" alt="Bike card" src={bikeCard} />
                </div>
            </div>
    );
}

export default MainPage;