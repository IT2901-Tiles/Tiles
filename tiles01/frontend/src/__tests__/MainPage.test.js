import {configure, shallow} from "enzyme"
import Adapter from 'enzyme-adapter-react-16';
import MainPage from '../Pages/MainPage';
import {cleanup} from '@testing-library/react';

afterEach(() => {
    cleanup();
});

configure({
    adapter: new Adapter()
});

describe('Animation page', () => {
    const wrapper = shallow(<MainPage/>)
    it('renders correctly', () => {
        shallow(<MainPage/>);
    });
    it('renders a LinkButton', () => {
        expect(wrapper.find('LinkButton')).toHaveLength(1);
    });
    it('on click gives these props', () => {
        wrapper.find('LinkButton').simulate('click');
        expect(wrapper.find('LinkButton').props()).toEqual({"size": "Large", "target": "/cards", "title": "Let's begin!"});
    });
    it('renders a p element', () => {
        const paragraph = wrapper.find('p');
        expect(paragraph.text()).toEqual('Welcome to the Tiles Internet of Things Invention! Letâ€™s create some smart creations! ');
        expect(paragraph).toHaveLength(1);
    });
});