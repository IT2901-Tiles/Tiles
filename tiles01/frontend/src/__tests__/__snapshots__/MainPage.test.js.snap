// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Animation page renders correctly 1`] = `
<div
  className="Page mainPage"
>
  <div>
    <img
      alt="The motion card"
      className="motionCardTop"
      src="motion.png"
    />
    <img
      alt="ColorChangeCard"
      className="colorChangeCardTop"
      src="color_change.png"
    />
    <img
      alt="Drop card"
      className="dropCardTop"
      src="drop.png"
    />
  </div>
  <div
    className="Logo"
    style={
      Object {
        "backgroundColor": "#E53D2FEE",
        "border": "0.25px solid #72140C",
        "borderRadius": "30px",
        "boxShadow": "1px 1px 3px #000000AA",
        "color": "#F3EAC2",
        "fontFamily": "Roboto",
        "textShadow": "2px 2px #252525ab",
      }
    }
  >
    <h1
      style={
        Object {
          "margin": "15px 0",
          "padding": "0",
          "position": "center",
        }
      }
    >
      TILES
    </h1>
  </div>
  <div
    className="infoBox"
  >
    <p>
      Welcome to the Tiles Internet of Things Invention! 
      <br />
      Letâ€™s create some smart creations! 
    </p>
  </div>
  <a
    href="/cards"
    onClick={[Function]}
  >
    <button
      className="LargeButton btn btn-primary"
      disabled={false}
      style={
        Object {
          "backgroundColor": "#F08A00",
          "color": "#F3EAC2",
        }
      }
      type="button"
    >
      Let's begin!
    </button>
  </a>
  <div>
    <img
      alt="The motion card"
      className="motionCard"
      src="motion.png"
    />
    <img
      alt="ColorChangeCard"
      className="colorChangeCard"
      src="color_change.png"
    />
    <img
      alt="Bike card"
      className="bikeCard"
      src="bike.png"
    />
  </div>
</div>
`;
